backend/unparsed_value.go

```diff
@@ -69,21 +69,15 @@ func ParseVariableValues(vv map[string]UnparsedVariableValue, decls map[string]*
 		if !declared {
 			switch val.SourceType {
 			case terraform.ValueFromConfig, terraform.ValueFromAutoFile, terraform.ValueFromNamedFile:
-				// These source types have source ranges, so we can produce
-				// a nice error message with good context.
-				//
-				// This one is a warning for now because there is an existing
-				// pattern of providing a file containing the superset of
-				// variables across all configurations in an organization. This
-				// is deprecated in v0.12.0 because it's more important to give
-				// feedback to users who make typos. Those using this approach
-				// should migrate to using environment variables instead before
-				// this becomes an error in a future major release.
-				if seenUndeclaredInFile < 3 {
+				// We allow undeclared names for variable values from files and warn in case
+				// users have forgotten a variable {} declaration or have a typo in their var name.
+				// Some users will actively ignore this warning because they use a .tfvars file
+				// across multiple configurations.
+				if seenUndeclaredInFile < 2 {
 					diags = diags.Append(tfdiags.Sourceless(
 						tfdiags.Warning,
 						"Value for undeclared variable",
-						fmt.Sprintf("The root module does not declare a variable named %q but a value was found in file %q. To use this value, add a \"variable\" block to the configuration.\n\nUsing a variables file to set an undeclared variable is deprecated and will become an error in a future release. If you wish to provide certain \"global\" settings to all configurations in your organization, use TF_VAR_... environment variables to set these instead.", name, val.SourceRange.Filename),
+						fmt.Sprintf("The root module does not declare a variable named %q but a value was found in file %q. If you meant to use this value, add a \"variable\" block to the configuration.\n\nIf you wish to provide certain \"global\" settings to all configurations in your organization, use TF_VAR_... environment variables to set these instead to silence this warning, or use the -compact-warnings option to show warnings in a compact way.", name, val.SourceRange.Filename),
```

Comment:
Minor suggestion for wording of the second paragraph to simplify reading and separate the two calls to action:

To silence these warnings, use TF_VAR_... environment variables to provide certain \"global\" settings to all configurations in your organization. To reduce the verbosity of these warnings use the -compact-warnings option.

Reply:
I like that suggestion and will adopt!